### ISSUE TITLE
confusing error message with typed macro on proc with semantic error

### ISSUE THREAD

**Issue by alaviss** [nim-works/nimskull#1108]
In this example, the error got shifted to the typed macro symbol within the proc pragma instead of being about the invalid symbol.

### Example

```nim
import macros

macro bar(x: typed): untyped =
  x

proc foobar() {.bar.} =
  totallyInvalidSym()
```

### Actual Output

```
test.nim(6, 17) Error: invalid pragma: bar
```

### Expected Output

```
test.nim(7, 3) Error: undeclared identifier: 'totallyInvalidSym'
```

### Additional Information

* Found during port of cps to nimskull
* The error can be found with a higher `--errorMax`

```
$ nim -v

Nimskull Compiler Version 0.1.0-dev.21160 [linux: amd64]

Source hash: 7e7226f1904360c2e14c8f57fc9e0f61e5483d58
Source date: 2024-01-19

active boot switches: -d:release
```

**Comment by saem** [comment]
Fixed as part of #1124, issue was not auto-closed.

**Comment by saem** [comment]
Fixed as part of #1124, issue was not auto-closed.

**Comment by saem** [comment]
Fixed as part of #1124, issue was not auto-closed.
